plejd/ble.bluez.js

```diff
@@ -393,16 +403,13 @@ class PlejdService extends EventEmitter {
         setTimeout(() => this.write(data, retry), 1000);
         return;
       }
-
       console.log('plejd-ble: write failed ' + err);
-      setTimeout(async () => {
-        await this.init();
+      await this.throttledInit(this.connectionTimeout * 1000);
```

Comment:
hmmm.. I just realized that this actually changed the entire write-flow. Since this might actually stop any writes at all until we have re-init'ed.
On the flip side: what sense does it make to try to write to a channel we know is broken?

Reply:
Agree, this looks suspicious. Since JS is single-threaded it's probably ok, but I would feel safer if we just put back the message in the write queue and spin of a new `init` without `await` (let the `write` function return)

Reply:
Yes, but it is quite important to keep the inits throttled / ”singletoned” because the current setup result in dbus errors presented by in #104

Reply:
True, but I'm thinking row 381 solves that?

Reply:
Will experiment with putting it back on the writeQ now and just let it return as your suggested.

Reply:
Nice. Maybe we mean the same thing, but just so I'm clear: I merely mean removing the "awaiting" on throttledInit, not remove the call completely.

Reply:
I know 👍 
The problem I am facing with that though.. is that for each "round" of runWriteQueue, we try to empty the queue before we allow us to timeout.. basically:
```
while q.length > 1
   if(error) q.push(data)
// => everlasting loop
```

So either write needs to flag fail / throw to be able to break the bad loop.. or keep current

Did find a few clearInterval-instead-of-clearTimeout bugs though which I will commit regardless of this convo.

Reply:
Ok, I see I actually have a new strategy for this (per-device queue, better retry strategy etc) in the works, so maybe it would be good to stop "as-is" with this PR and then see where we are. Main thing to me is that each PR doesn't degrade existing functionality. We don't need to solve all problems at once ;)

Reply:
This is my final atempt in that case. Basically rewriting who is responsible for inits:
https://github.com/JohnPhoto/hassio-plejd/pull/5

Feel free to take a look and see which one you prefer (That PR is targeting this PR to highlight the diffs agains work already done here)

Reply:
I prefer this one compared to the JohnPhoto#5 PR. I think `retryWriteQueue` is a nice idea but solves the problem possibly in the wrong way. I have experimented quite a bit with transitions, retries, and so on and come to the conclusion that you typically only want to send the latest command to each queue (if you set brightness 1 => 50 => 255 you don't want 1 to be retried).

So I suggest merging this and I'll see if I can fix the bug I have when transitioning to off (light sometimes step up brightness before dimming down). WIP is here if you want to look at the idea. It currently implements "shouldRetry" on each command, but could easily be extended to a retryCount > 1. Not rewritten after improvements in this PR obviously :)

Reply:
Fair enough. I just started with this yesterday so I’m new here 😀

Reply:
I beat you to this project with a few days then 😉